pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/PublicSettings.scala

```diff
@@ -0,0 +1,143 @@
+ */
+class PublicSettings(domain: String, callback: Try[Map[String, String]] => Unit = _ => (), scheduler: Scheduler = StdSchedulerFactory.getDefaultScheduler())
+                    (implicit client: Http, ec: ExecutionContext) {
+
+  private val agent = Agent[Map[String, String]](Map.empty)
+  private val job = JobBuilder.newJob(classOf[FunctionJob])
+    .withIdentity(s"refresh-public-key-$domain")
+    .build
+
+  def start(intervalInSeconds: Int = 60) = {
+    // create job shedule
+    val schedule = SimpleScheduleBuilder.simpleSchedule
+      .withIntervalInSeconds(intervalInSeconds)
+      .repeatForever()
+    val trigger = TriggerBuilder.newTrigger()
+      .withSchedule(schedule)
+      .build
+    // add this job's action to the global map of currently running jobs
+    PublicSettings.jobs.put(job.getKey, () => refresh())
+    // shedule the job (stopping current one if it is already there)
+    if (scheduler.checkExists(job.getKey)) {
+      scheduler.deleteJob(job.getKey)
+    }
+    scheduler.scheduleJob(job, trigger)
+    // ensure the scheduler is running
+    scheduler.start()
+  }
+  def stop() = {
+    // remove job from map of current jobs and deschedule
+    PublicSettings.jobs.remove(job.getKey)
+    scheduler.deleteJob(job.getKey)
+  }
+  def refresh() = {
+    val publicFuture = PublicSettings.getPublicSettings(domain)
+    publicFuture
+      .onComplete(callback)
+    publicFuture
+      .foreach(settings => agent.send(settings))
+  }
+
+  def publicKey = agent.get().get("publicKey")
+  val bucketName = PublicSettings.bucketName
+  val cookieName = PublicSettings.cookieName
+  val assymCookieName = PublicSettings.assymCookieName
+}
+
+/**
+ * Static PublicSettings for applications that do not want to use the provided mechanism for auto-refreshing
+ * public data.
+ */
+object PublicSettings {
+  val bucketName = "pan-domain-auth-settings"
+  val cookieName = "gutoolsAuth"
+  val assymCookieName = s"$cookieName-assym"
+
+  def getPublicSettings(domain: String)(implicit client: Http, ec: ExecutionContext): Future[Map[String, String]] = {
+    fetchSettings(domain) flatMap extractSettings
+  }
+
+  /**
+   * Fetches the public key from the public S3 bucket
+   *
+   * @param domain the domain to fetch the public key for
+   * @param client implicit dispatch.Http to use for fetching the key
+   * @param ec     implicit execution context to use for fetching the key
+   */
+  def getPublicKey(domain: String)(implicit client: Http, ec: ExecutionContext): Future[String] = {
+    getPublicSettings(domain) flatMap extractPublicKey
+  }
+
+  // internal functions for fetching and parsing the responses
+  private def fetchSettings(domain: String)(implicit client: Http, ec: ExecutionContext): Future[Either[Throwable, String]] = {
+    val req = host("s3-eu-west-1.amazonaws.com").secure / bucketName / s"$domain.settings.public"
+    client(req OK as.String).either
```

Comment:
As @theefer says, no. This function takes the domain as an argument because the public key varies by domain (i.e. the environment). The naming convention is currently `<domain>.settings`, I'm adding a corresponding public file at `<domain>.settings.public`.